### PR TITLE
Add request to fetch latest user packages

### PR DESCRIPTION
**Pull request by zeusongit**
### Purpose

Add route to fetch all latest package versions from the logged-in user, with their scan results.

### Declarations

Check these if you believe they are true

- [x] The codebase is in a better state after this PR
- [ ] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [ ] The level of testing this PR includes is appropriate
- [ ] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions) and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.

### Reviewers

@DynamoDS/dynamo

### CODE REVIEW

File: src/GregClient/Requests/GetUserPackages.cs [hunk 1]
```diff
@@ -0,0 +1,30 @@
+﻿using RestSharp;
+
+namespace Greg.Requests
+{
+    public class GetUserPackages : PackageReferenceRequest
```
Comment (QilongTang) [inline src/GregClient/Requests/GetUserPackages.cs:5]: comments please

**Comment by QilongTang** [comment]
Also this turns out to be a new file, no need to touch csproj file?

**Comment by zeusongit** [comment]
> Also this turns out to be a new file, no need to touch csproj file?

I was also expecting a change in csproj file, I checked and there is no reference for any such classes in there

File: src/GregClient/Requests/GetUserPackages.cs (hunk 1 above)
Comment (QilongTang) [reply src/GregClient/Requests/GetUserPackages.cs:8]: Maybe also comments for all public properties

File: src/GregClient/Requests/GetUserPackages.cs [hunk 2]
```diff
@@ -0,0 +1,46 @@
+﻿using RestSharp;
+
+namespace Greg.Requests
+{
+    /// <summary>
+    /// Request for getting latest versions of all packages published by the current user.
+    /// </summary>
+    public class GetUserPackages : PackageReferenceRequest
```
Comment (mjkkirschner) [inline src/GregClient/Requests/GetUserPackages.cs:8]: Are there other `User` routes that don't actually take a user object? What do you think about calling this GetMyPackages?

File: src/GregClient/Requests/GetUserPackages.cs [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+    {
+        private readonly Method httpMethod = Method.GET;
```
Comment (mjkkirschner) [inline src/GregClient/Requests/GetUserPackages.cs:10]: not a big deal, but feels kind of unnecessary.

**Review by mjkkirschner — APPROVED** [review]

**Review by QilongTang — APPROVED** [review]

**Pull request merged**